lib/peer-discovery.js

```diff
@@ -22,6 +28,20 @@ module.exports = class PeerDiscovery {
     this.refresh().catch(this._onerror)
   }
 
+  session ({ server = true, client = true }) {
+    if (this.destroyed) throw new Error('PeerDiscovery is destroyed')
+
+    if (client) this._clientSessions++
+    if (server) this._serverSessions++
+
+    const session = new PeerDiscoverySession(this, { server, client })
+    this._sessions.push(session)
+
+    this.refresh().catch(safetyCatch)
```

Comment:
use an onerror hook like above

Reply:
which we should default to safetyCatch everywhere :)

Reply:
All set!